### ISSUE TITLE
`react/no-unused-prop-types` false positives with props closure

### ISSUE THREAD

**Issue by a-x-** [yannickcr/eslint-plugin-react#2171]
it's similar to https://github.com/yannickcr/eslint-plugin-react/issues/2155, but w/o passing via arguments.

I created bug-repro-repo: https://github.com/a-x-/eslint-memo-props-check

In my case props passed as closure w/o function arguments
brief:
```jsx
function Foo (props) {
  // setup hooks: useState,,,
  return <div>{ renderPhoto() }</div>;

  function renderPhoto () {
    return <div>{ props.renderPhotoTools() }</div>;
  }
  // handlers
}
```

**Comment by ljharb** [comment]
This is very strange - why would you want to create a new `renderPhoto` function on every render of `Foo`, rather than hoisting it outside?

Either way, this should indeed work.

**Comment by a-x-** [comment]
because of closure and multiple variables.
anyway, I'll think about you point too, thank you
